app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsPresenter.kt

```diff
@@ -27,8 +35,19 @@ class DebugSettingsPresenter(
         view?.showSettings(settings)
     }
 
-    override fun onNetworkChanged(newName: String) {
-        this.networkName = newName
+    override fun onNetworkChanged(newNetworkEnvironment: NetworkEnvironment) {
+        this.networkName = newNetworkEnvironment.name
+
+        launch {
+            try {
+                environmentManager.chooseEnvironment(newNetworkEnvironment)
```

Comment:
is any of calls can throw exception? As i remember, it will not

Reply:
RenVMService.stopService(context) can
and homeLocalRepository.clear() - too, it's working with DB